lib/global-styles.php

```diff
@@ -171,22 +171,32 @@ function gutenberg_experimental_global_styles_get_theme() {
  * @return string CSS rule.
  */
 function gutenberg_experimental_global_styles_resolver( $global_styles ) {
-	$css_rule = '';
-
-	$token    = '--';
-	$prefix   = '--wp' . $token;
-	$css_vars = gutenberg_experimental_global_styles_get_css_vars( $global_styles, $prefix, $token );
-	if ( empty( $css_vars ) ) {
-		return $css_rule;
-	}
+	$css_rules = '';
+
+	// This is only for demo purposes, it should be pluggable
+	// so new selectors can be register
+	// (3rd party blocks, custom selectors, etc).
+	$selectors = array(
```

Comment:
This mapping should be dynamic. Blocks could declare their own selectors (ex: `core/paragraph`) but also themes may want to register selectors (ex: `custom/paragraph-in-quote`). I'm not certain yet of what option(s) would be best in terms of implementation, how these should be registered. This can be investigated later if the direction here seems worth diving deeper.

Reply:
Regarding this type of change, is the intention of global styles to target very specific elements like this `.wp-block-quote p`? How specific should this type of selector get, and where does it cross into the territory of the normal style.css file?

Reply:
What I like about this proposal is that it presents a simple and familiar mindset to think about the style system. It also leverages how CSS works natively. We're giving blocks hook points (CSS variables) that authors can plug into. The selectors we allow have the same reach as the hooks we introduce: authors are able to tweak all the existing hooks, whether they target the whole (core), all blocks of a kind (core/paragraph), or only parts of them (cite within a quote block).

It's a bit like WordPress hooks, but for CSS. As with PHP or JavaScript, you can't do all the things with hooks. The theme's stylesheet will remain relevant. The frontier is a bit blurry at this stage, though.

Reply:
@nosolosw Thanks for clarifying that for me! It sounds like the desire is to allow theme authors to be able to use global styles for anything they want, and it really just depends on how much control they want to give their users, and how many knobs-and-switches you want to end up with in the site editor. Does that sound right?